helm/provisioner/templates/provisioner.yaml

```diff
@@ -64,7 +64,16 @@ spec:
         app: local-volume-provisioner
     spec:
       serviceAccountName: {{.Values.daemonset.serviceAccount}}
+{{- if .Values.daemonset.priorityClassName }}
+{{- if eq .Values.common.namespace "kube-system" }}
       priorityClassName: {{.Values.daemonset.priorityClassName}}
+{{- else }}
+      {{- $systemCriticalClasses := list "system-cluster-critical" "system-node-critical" }}
+      {{- if not (has .Values.daemonset.priorityClassName $systemCriticalClasses) }}
+      priorityClassName: {{.Values.daemonset.priorityClassName}}
```

Comment:
should we have a test case where you can set a non-critical priority class?

Reply:
Updated, two examples are added in this PR:

- helm/examples/baremetal-priority-critical.yaml
- helm/examples/baremetal-priority-noncritical.yaml

Generated manifest files are tested manually. (try to test against cluster automatically in #64)